src/pages/Editor/containers/editContext.js

```diff
@@ -11,10 +11,69 @@ const svgStyles = {
   top: 0,
   left: 0,
 };
-const EditContextProvider = props => {
+const EditContextProvider = (props) => {
   const aImagePrefix = "";
   const [pageSrc, setPageSrc] = useState(`${aImagePrefix}blank1.png`);
   const [isBody, setIsBody] = useState(true);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pages, setPages] = useState([
+    {
+      head: {
+        text: "",
+        headLeft: 0,
+        headTop: 0,
+        headWidth: null,
+      },
+      body: {
+        text: "",
+        bodyLeft: 0,
+        bodyTop: 0,
+        bodyWidth: null,
+      },
+    },
+  ]);
+
+  useEffect(() => {
+    if (currentPage === pages.length) {
+      setCurrentPage((currentPage) => currentPage - 1);
+    }
+  });
```

Comment:
```suggestion
  }, [currentPage]);
```
@niteshsinha17 I suppose we don't want this to render every time there is a state change. Rather, we want this to re-render once the `currentPage` gets updated.

In any case, it's better to add a dependency array to this useEffect hook and make it render conditionally rather than on each render.